src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/IO/Packaging/PreloadedPackages.cs

```diff
@@ -237,8 +213,8 @@ internal bool ThreadSafe
         // ListDictionary is the best fit for this scenarios; otherwise we should be using
         // Hashtable. HybridDictionary already has functionality of switching between
         //  ListDictionary and Hashtable depending on the size of the collection
-        static private HybridDictionary _packagePairs;
-        static private readonly Object  _globalLock;
+        static private Dictionary<Uri, PackageThreadSafePair> _packagePairs;
```

Comment:
Sorry, I do not think replace the HybridDictionary with `Dictionary<Uri, PackageThreadSafePair>` is a good idea. Please see the comment:

> ListDictionary is the best fit for this scenarios; otherwise we should be using Hashtable. HybridDictionary already has functionality of switching between ListDictionary and Hashtable depending on the size of the collection